### PR TITLE
[VENTUS][fix] Add function sections for workitem functions

### PR DESCRIPTION
**Pull request by zhoujingya**
By adding function sections for workitem functions, the unused functions can be removed by linker to reduce binary code size

### CODE REVIEW

**Comment by Jules-Kong** [comment]
是否可以类似于这样
`

.section .text,"ax",@progbits   
.global __builtin_riscv_num_groups_x   
....   
....   
ret    

.global __builtin_riscv_num_groups_y   
.type __builtin_riscv_num_groups_y, @function   
....   
....   
ret    
`

**Comment by zhoujingya** [comment]
The check failure is caused by pocl change as mentioned above

**Comment by Jules-Kong** [comment]
Have you tested the basic test cases of cts? @zhoujingya

**Comment by zhoujingya** [comment]
> Have you tested the basic test cases of cts? @zhoujingya

@Jules-Kong It will not afftect the CTS test cases, it only affects the binary code size, but we can run the test later

**Comment by Jules-Kong** [comment]
> > Have you tested the basic test cases of cts? @zhoujingya
> 
> @Jules-Kong It will not afftect the CTS test cases, it only affects the binary code size, but we can run the test later

@zhoujingya Theoretically it has no impact, but testing is a necessary means.

**Comment by Jules-Kong** [comment]
@zhoujingya Please merge the two commits into one.

File: README.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ Use command line under the root directory of `llvm-ventus`
 Linking CRT and libclc
 All the libclc workitem is included in `riscv32clc.o`
 ```sh
-./install/bin/ld.lld -o vecadd.riscv -T utils/ldscripts/ventus/elf32lriscv.ld vecadd.o ./install/lib/crt0.o ./install/lib/riscv32clc.o -L./install/lib -lworkitem --gc-sections
+./install/bin/ld.lld -o vecadd.riscv -T utils/ldscripts/ventus/elf32lriscv.ld vecadd.o ./install/lib/crt0.o ./install/lib/riscv32clc.o ./install/lib/workitem.o --gc-sections
```
Comment (Jules-Kong) [inline README.md:127]: Would it be better to make it a dynamic link directly? In this way, there is no need to consider the issue of gc-section. The codesize will become smaller, but there may be a 1%-2% loss in performance.

File: libclc/riscv32/lib/workitem/workitem.S [hunk 2]
```diff
@@ -353,7 +353,7 @@ __builtin_riscv_global_offset_z:
   ret
 
 
-  .text
+  .section	.text.__builtin_riscv_num_groups_x,"ax",@progbits
```
[the remark this answers is not in the document]
Comment (zhoujingya) [reply libclc/riscv32/lib/workitem/workitem.S:356]: yesy

File: README.md (hunk 1 above)
Comment (zhoujingya) [reply README.md:127]: No dynamic link since dynamic library require operating system

**Pull request merged**